src/widget/text.rs

```diff
@@ -12,44 +12,81 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use parley::Layout;
+use parley::{FontContext, Layout};
 use vello::{
-    kurbo::{Affine, Point, Size},
+    kurbo::{Affine, Size},
     peniko::{Brush, Color},
-    SceneBuilder, SceneFragment,
+    SceneBuilder,
 };
 
 use crate::text::ParleyBrush;
 
 use super::{
-    align::{FirstBaseline, LastBaseline, SingleAlignment, VertAlignment},
-    contexts::LifeCycleCx,
-    AlignCx, ChangeFlags, EventCx, LayoutCx, LifeCycle, PaintCx, RawEvent, UpdateCx, Widget,
+    contexts::LifeCycleCx, BoxConstraints, ChangeFlags, Event, EventCx, LayoutCx, LifeCycle,
+    PaintCx, UpdateCx, Widget,
 };
 
 pub struct TextWidget {
     text: String,
     layout: Option<Layout<ParleyBrush>>,
-    is_wrapped: bool,
 }
 
 impl TextWidget {
     pub fn new(text: String) -> TextWidget {
-        TextWidget {
-            text,
-            is_wrapped: false,
-            layout: None,
-        }
+        TextWidget { text, layout: None }
     }
 
     pub fn set_text(&mut self, text: String) -> ChangeFlags {
         self.text = text;
         ChangeFlags::LAYOUT | ChangeFlags::PAINT
     }
+
+    fn get_layout_mut(&mut self, font_cx: &mut FontContext) -> &mut Layout<ParleyBrush> {
+        // Ensure Parley layout is initialised
+        if self.layout.is_none() {
+            let mut lcx = parley::LayoutContext::new();
+            let mut layout_builder = lcx.ranged_builder(font_cx, self.text.trim(), 1.0);
```

Comment:
Not sure if we want to trim strings, having spaces at the beginning or the end of a string may be intended.